bundles/org.openhab.core.transform/src/main/java/org/openhab/core/transform/Transformation.java

```diff
@@ -12,39 +12,43 @@
  */
 package org.openhab.core.transform;
 
+import java.util.Map;
 import java.util.Objects;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.core.common.registry.Identifiable;
 
 /**
- * The {@link TransformationConfiguration} encapsulates a transformation configuration
+ * The {@link Transformation} encapsulates a transformation configuration
  *
  * @author Jan N. Klug - Initial contribution
  */
 @NonNullByDefault
-public class TransformationConfiguration implements Identifiable<String> {
+public class Transformation implements Identifiable<String> {
+    public static final String FUNCTION = "function";
+
     private final String uid;
     private final String label;
     private final String type;
     private final @Nullable String language;
```

Comment:
I would claim that we do not need a `language` property anymore - all language specific stuff is within the `configuration` and those are localizable by default already. The `Transformation` instance itself should not be dependent on a single language.

Reply:
Well, it has always been this way for text-based transformations. I'm not sure we'll break something if the leave it out.

Reply:
Could you please elaborate on this? I'm not sure I understand where we have a language dependency in text-based transformations.

Reply:
This has always been the case: https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.transform/src/main/java/org/openhab/core/transform/AbstractFileTransformationService.java. It's implemented by two characters separated with a `_` before the file extension.

Reply:
Interesting, I've never heard about this feature and it does not turn up in any documentation either.
But looking at it, I think it is something different than saying a `Transformation` is language specific. Instead, it brings multi-language support to a single `Transformation` - in your example, the transformation is a MAP with the parameter `switch` and the logic then fetches the right content based on the currently set locale. I.e. all files (`switch.map`, `switch_de.map` and `switch_fr.map`) belong to the very same `Transformation`.

Reply:
I‘m not sure how long this feature has been there, looking at the commit history at least since 2.5 and I doubt that many people use it. And BTW I don‘t think it‘s working as intended.

I would think that the use-case would be to use it in the UI to display the switch status in different languages (would be very useful for installations with international guests). But then it should select the language based on the browser locale, but this is not the case, it‘s the openHAB system‘s locale that is used.

I agree that it is more „one transformation, different languages“, but it seems to be very difficult to find a way to have different languages in one transformation when configured from UI. We would need to add different functions for each language, that makes the UI very complex.

I think the way it is now is a better compromise between „good modeling“, „backward compatibility“ and „usability from UI“.